Remove the _check_imas_version() call and helper, which dereference
imas.backends.imas_core.imas_interface.has_imas -- removed for good in
IMAS-Python 2.3.0, so every validate() call raises AttributeError.
Backport of imas-validator develop commits 04050dc + 0adacdf.
author: Yannick de Jong (ITER PDS)
--- imas_validator-1.0.0.orig/imas_validator/validate/validate.py	2026-08-18 10:36:47.901337589 +0200
+++ imas_validator-1.0.0/imas_validator/validate/validate.py	2026-08-18 10:36:47.917708780 +0200
@@ -6,7 +6,6 @@
 import sys
 
 import imas  # type: ignore
-from packaging.version import Version
 
 # from imas_validator.exceptions import IMASVersionError
 from imas_validator.rules.loading import load_rules
@@ -35,7 +34,6 @@
         List of IDSValidationResult objects
     """
 
-    _check_imas_version()
     try:
         dbentry = imas.DBEntry(imas_uri, "r")
     except (imas.exception.ALException, imas.exception.LowlevelError) as e:
@@ -58,31 +56,3 @@
     dbentry.close()
     return results_collection
 
-
-def _check_imas_version() -> None:
-    """Check if the installed IMAS version is sufficient."""
-    # TODO: check if this is the best level to test for the IMAS version
-    if not imas.backends.imas_core.imas_interface.has_imas:
-        version_found = imas.backends.imas_core.imas_interface.ll_interface._al_version
-        if version_found is None:
-            logger.info(
-                "No IMAS install could be found."
-                "IDS Validation will work with limited functionality."
-            )
-            return
-        else:
-            logger.info(f"Found IMAS install with version {version_found}.")
-
-    if imas.backends.imas_core.imas_interface.ll_interface._al_version < Version("5.1"):
-        logger.info(
-            "IDS Validation requires an IMAS installation of version 5.1 or newer."
-            "See the README for more details."
-        )
-
-    # if not imas.backends.imas_core.imas_interface.has_imas:
-    #     raise IMASVersionError()
-    # if imas.backends.imas_core.imas_interface.ll_interface._al_version
-    # < Version("5.1"):
-    #     raise IMASVersionError(
-    #         imas.backends.imas_core.imas_interface.ll_interface._al_version
-    #     )
